src/Show.js

```diff
@@ -3,13 +3,41 @@ import "./Show.css";
 
 const tumblr_re = /^\d+?\.media\.tumblr\.com$/;
 
+const getGfycatUrl = urlPathname => {
```

Comment:
i'm thinking that this function could better be implemented as a singleton with closure so that the url store doesn't need to be retrieved from Storage on every call. Maybe something like:
```
const getGfycatUrl = (() => {
  const fromStorage = sessionStorage.getItem("gfycatStore")
  const store = fromStorage ? JSON.parse(fromStorage) : {}
  return async (gfyId) => { // can also use function that returns Promise
    const entry = store[gfyfId]
    // TODO: write to `store` and also write to sessionStorage
  }
})()
```

Reply:
hm my intention here wasn't to create another object that would be used by your `getGfyatUrl` function, but rather to have `getGfycatUrl` handle the closure itself. The current iteration seems like a step backwards in terms of code semantics and usability/maintainability. I'm considering just closing this pr and implementing the changes myself since I don't want to waste any more time with potential miscommunications.